core/include/ikos/core/domain/memory/value.hpp

```diff
@@ -1162,17 +1187,115 @@ class ValueDomain final
     return updated_cells;
   }
 
+
+
+  /// \brief Detect whether the read bits are initialized, uninitialized,
+  /// for possibly uninitialized.
+  ikos::core::Uninitialized is_read_uninitialized(MemoryLocationRef base,
+                            const MachineInt& offset,
+                            const MachineInt& size) {
+    if (!size.fits<uint32_t>()) {
+      // The ZNumber infrastructure cannot handle shifts by
+      // more than the max 32-bit number, so give up in that case.
+      // and assume possibly uninitialized.
+      return ikos::core::Uninitialized::top();
+    }
+
+    // Use ZNumbers as bitvectors for bits [0,size) relative to the
+    // low bound of the read (i.e. offset).
+    ZNumber zoffset = offset.to_z_number();
+    ZNumber zsize = size.to_z_number();
+    ZNumber zupper = zoffset + zsize;
+    ZNumber read_mask = make_clipped_mask(zoffset, zsize, zoffset, zsize);
+
+    CellSetT cells = this->_cells.get(base);
+    ZNumber initialized_coverage = ZNumber(0);
+    ZNumber uninitialized_coverage = ZNumber(0);
+    for (VariableRef cell : cells) {
+      const ZNumber& other_offset = CellVariableTrait::offset(cell).to_z_number();
+      const ZNumber& other_size = CellVariableTrait::size(cell).to_z_number();
```

Comment:
`to_z_number` returns a fresh `ZNumber` so you should take a copy.
```suggestion
      ZNumber other_offset = CellVariableTrait::offset(cell).to_z_number();
      ZNumber other_size = CellVariableTrait::size(cell).to_z_number();
```
NB: I know c++ probably does the right thing even with `const ZNumber&` (i.e, extend the lifetime of the ZNumber) here but I wouldn't rely on it.

Reply:
Will do.